### PR TITLE
Development

### PR DESCRIPTION
**Pull request by serdg0**

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require './lib/tic-tac-toe'
```
Comment (mosaaleb) [inline spec/game_spec.rb:3]: - [ ] All the public methods of the classes implemented in project OOP are tested
There are some method in Game class are not tested yet `#turn`, `#turn_count`, `#draw`, `#get_input`, `#not_repeated` .... 
Let's add them.

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+RSpec.describe Game do
```
Comment (mosaaleb) [inline spec/game_spec.rb:5]: - [x] To define a variable, you need to use `let`
Why it's different from creating instance variables? `let` uses lazy evaluation means that if the variable won't be created unless it is invoked
So `let(:game) { Game.new }` won't be created unless you references `game` inside an example.

Here are some references to read more about:
- [let vs instance variable](https://mixandgo.com/learn/let-vs-instance-variables-in-rspec)
- [when to use let vs instance](https://stackoverflow.com/questions/14116613/when-is-using-instance-variables-more-advantageous-than-using-let)

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    describe "#check_winner?" do
+
+        it "returns true if top row is the winner" do
+            board = Board.new
+            player = Player.new
+            player.color = "X"
+            board.return_board_input(1,player)
+            board.return_board_input(2,player)
+            board.return_board_input(3,player)
```
Comment (mosaaleb) [inline spec/board_spec.rb]: - **`[OPTIONAL]`** There is a convention in rspec that if you are testing specific functionality don't use another methods in that test. For example here we are testing `check_winner` and we are using another method `return_board_input`, now what if this other method is buggy and the test fails? You will waste alot of time trying to fix `check_winner`, but the other method is the one you need to fix.

To fix this we can manually stub the board cells with our desired values.

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Well done @AlejoCode, @serdg0 :clap:

### All test cases are green :rocket:, however there is some requirements needs to be handled before moving on.

- [ ] There is alot of errors related to styling guidelines

Screenshot from 2019-08-10 00-10-45

Please, take a look at [guidelines](https://github.com/rubocop-hq/ruby-style-guide) and  let's fix these errors. You also need to use stickler with rubocop specified inside `.stickler.yml` [stickler](https://github.com/microverseinc/linters-config/tree/master/ruby)

**TIP** You can install rubocop extension available on every text editor to catch these erros while you wrote them, before even pushing.

- [x] Let vs static variables. Please see my [comment](https://github.com/serdg0/tic-tac-toe_rspec/pull/1#discussion_r312659650).

**Friendly reminder:** You should use Git Flow in this project. However, changing it now would take too much time. Please make sure you use Github/Git Flow in the next project. You will find instructions in the Path-wright lesson description.

Please fix these minor requirements and submit a new code review.

### Happy Coding :computer:

**Comment by bellom** [comment]
## Currently Reviewing this PR

**Review by bellom — CHANGES_REQUESTED** [review]
Hi @serdg0 @AlejoCode, 
Well done guys, according to the previous review, you still need to work on those changes mentioned by the reviewer, you only two test case inside the `game_spec`.
And am not seeing linter check on this project, but previous reviewer talk about it, i hope am not missing something....kindly add the linter and fix all errors if need too, checkout this [link](https://github.com/microverseinc/linters-config#troubleshooting) for help.
kindly re-submit for review after making those changes....

**Comment by bellom** [comment]
Currently reviewing this PR

**Review by bellom — APPROVED** [review]
Hi Team,
Well done guys.
Kindly know that it's part of the project requirement to add linter and be sure to use `Git Flow` when necessary.
**Go head merge :+1:** 
Goodluck in your next project.

**Pull request merged**